### PR TITLE
Model Validations

### PR DESCRIPTION
**Pull request by zofskeez**
After upgrading to Ember 3.28 there were issues with the `ember-cp-validations` addon. This PR provides an internal implementation for model validations and removes the addon.

### CODE REVIEW

File: ui/app/components/secret-edit-metadata.js [hunk 1]
```diff
@@ -54,6 +54,7 @@ export default class SecretEditMetadata extends Component {
     if (value) {
       if (name === 'customMetadata') {
         // cp validations won't work on an object so performing validations here
+        // TODO: review this and incorporate into model-validations system
```
Comment (Monkeychip) [inline ui/app/components/secret-edit-metadata.js:57]: Can you put your initials in front of the TODO? e.g. ARG TODO or CBS TODO (you'll see those examples in front of some in the codebase). This way you can search for your specific TODOs

File: ui/app/services/path-help.js [hunk 2]
```diff
@@ -298,8 +299,17 @@ export default Service.extend({
           // Build and add validations on model
           // NOTE: For initial phase, initialize validations only for user pass auth
           if (backend === 'userpass') {
-            let validations = buildValidations(fieldGroups);
-            newModel = newModel.extend(validations);
+            const validations = fieldGroups.reduce((obj, element) => {
```
Comment (Monkeychip) [inline ui/app/services/path-help.js:302]: nice!

**Review by Monkeychip — APPROVED** [review]
Great work!

File: ui/app/components/secret-edit-metadata.js (hunk 1 above)
Comment (zofskeez) [reply ui/app/components/secret-edit-metadata.js:57]: For sure thanks for the tip!

**Pull request merged**